newrelic/resource_newrelic_alert_muting_rule.go

```diff
@@ -41,7 +43,23 @@ func resourceNewRelicAlertMutingRule() *schema.Resource {
 									"attribute": {
 										Type:         schema.TypeString,
 										Required:     true,
-										ValidateFunc: validation.StringInSlice([]string{"accountId", "conditionId", "policyId", "policyName", "conditionName", "conditionType", "conditionRunbookUrl", "product", "targetId", "targetName", "nrqlEventType", "tag", "nrqlQuery"}, false),
+										ValidateFunc:func(val interface{}, key string) (warns []string, errs []error) {
+											valueString := val.(string)
+											attemptedTagRegex := regexp.MustCompile(`^tag`)
+											correctTagRegex := regexp.MustCompile(`^tag\..+$`)
+
+											// tag.SomeValue attempted but does not match allowed format
+											if attemptedTagRegex.Match([]byte(valueString)){
+												if !correctTagRegex.Match([]byte(valueString)){
+													errs = append(errs, fmt.Errorf("%#v of %#v must be in the format tag.tag_name", key, valueString))
+													return
+												} else {
+													return
```

Comment:
I think the `else` condition/return can be removed so it continues to fallback to the general validation after.

Reply:
I believe it will fail because `tag.whatEver` won't be part of the slice of strings it's validating against, so I think we need to return here.

I left "tag" in the list for completeness' sake in the error message, but given this check and short circuit, it shouldn't ever let a plain "tag" through (without the `.tagName` suffix)

Reply:
😅 I just needed to outdent the return statement. To reiterate, we still want this return, since we've detected and validated a `tag.Whatever` and want to skip checking against the slice of fixed values.